x-pack/legacy/plugins/snapshot_restore/public/app/services/http/use_request.ts

```diff
@@ -61,20 +60,34 @@ export const useRequest = ({
   body,
   interval,
   initialData,
-  timeout,
   uimActionType,
 }: UseRequest) => {
+  // Main states for tracking request status and data
   const [error, setError] = useState<null | any>(null);
   const [loading, setLoading] = useState<boolean>(true);
   const [data, setData] = useState<any>(initialData);
 
+  // States for tracking polling
+  const [polling, setPolling] = useState<boolean>(false);
+  const [currentInterval, setCurrentInterval] = useState<UseRequest['interval']>(interval);
+  const intervalRequest = useRef<any>(null);
+  const isFirstRequest = useRef<boolean>(true);
```

Comment:
Do you think we could return `isFirstRequest.current` as part of the return object, and the UI could just use that to determine whether or not to show the loading spinner? That would mean we wouldn't need the `polling` state or `isPollRequest` logic in this hook.

If that's possible then I think that'd be an improvement because then the UX would be clearer when reading the consuming code _and_ this hook would become more reusable. I'd love to extract this hook so we can reuse it in all of our apps and I can think of situations where we may want to show a subtle spinner when polling (e.g. when refreshing Console autocomplete definitions).

Reply:
I will defer any changes to this hook for when we do re-use it outside this app. as it currently is, the recovery status table _does_ show a subtle spinner (next to interval dropdown) when a poll request is in progress by using the current returned `polling` state